src/SDK.ts

```diff
@@ -374,6 +385,10 @@ export default class SDK implements ISDK {
     headers[OmnichannelHTTPHeaders.authenticatedUserToken] = authenticatedUserToken;
     headers[OmnichannelHTTPHeaders.authCodeNonce] = this.configuration.authCodeNonce;
 
+    if (this.sessionId) {
```

Comment:
this block is repeated 12 times , any way we can avoid the repetition?

Reply:
Refactored